pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -197,7 +202,29 @@ def execute(
             self.tracker.update(batches=1, executions=len(circuits))
             self.tracker.record()
 
-        results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        if execution_config.max_workers is None:
+            max_workers = self._max_workers
+        else:
+            max_workers = execution_config.max_workers
+
+        if max_workers is None:
+            results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        else:
+            import cloudpickle  # pylint: disable=import-outside-toplevel
```

Comment:
Is use of `cloudpickle` a hard requirement, or more of a convenience?

Reply:
It is there because `pickle` is limited in what it can pickle. For instance, as soon as there is a lambda function somewhere in `simulate` or its arguments, the multiprocessing map will fail because it cannot pickle before sending to subprocesses. For now, I took the easy way out to replace the Pickler with one that has a more complete dispatch table.